### PR TITLE
Continue to draw when input events freeze

### LINKED ISSUE #913: Window redraws can pause during resize
Related to #12, the runloop pauses on move.resize on some systems.

on MacOS: resize blocks the runloop
on Windows: resize and move block the runloop

In #531 we added live refresh of canvas, but this only works when the window is actually moved. If the user is holding down the resize action but not making a resize it does not repaint (like on a move).

Therefore we need to run a background timer to simulate the runloop for macOS and Windows when we detect that it has been paused (because GLFW is blocking on event poll).

### PR DESCRIPTION
**Pull request by andydotxyz**
### Description:
So that we can continue to paint when window is moving or resizing we need to run on a separate thread.
We spawn a goroutine that manages drawing, like the current main loop manager.
Add another goroutine that triggers a window refresh of any dirty window every 60 FPS (keep things clean).
Windows can request a redraw by being pushed to the queue, like the canvas system.

Fixes #913 

### Checklist:

- [ ] Tests included. <- still not found a way
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

**Comment by andydotxyz** [comment]
Travis passed :) seems that draft PRs still have an issue with getting the data back

File: internal/driver/glfw/glfw_test.go [hunk 1]
```diff
@@ -2,8 +2,15 @@
...
+import "time"
+
 func repaintWindow(w *window) {
-	runOnMain(func() {
+	for !running() {
+		// Wait for GLFW loop to be running.
+		// If we try to create windows before the context is created, this will fail with an exception.
+		time.Sleep(time.Millisecond * 10)
+	}
+	runOnDraw(w, func() {
```
[the remark this answers is not in the document]
Comment (andydotxyz) [reply internal/driver/glfw/glfw_test.go:13]: This is slightly different. We want it to run as soon as possible, but it's can't be run on the wrong thread or GL might get associated with the wrong run context.
I think the comment may be a copy-paste issue

[the remark this answers is not in the document]
Comment (andydotxyz) [reply internal/driver/glfw/glfw_test.go:13]: And this is just a test, so I don't want to make production functionality just to support this ;)

**Comment by andydotxyz** [comment]
Is anyone else seeing red flashes when resizing fyne_demo on the "Windows" screen?
That is weird!

**Comment by andydotxyz** [comment]
Somehow the first 3 commits on this branch have ended up on develop. Which is super strange as it's in my fork.
Looks like I did something bad :( Need to get this wrapped up so I can merge it in and fix develop which is in a partial state right now...

**Comment by andydotxyz** [comment]
The red flashes seem to happen on screens that have not re-calculated their layout before the frame is drawn during resize. 
Maybe Cedric's redraw patch helps here? not too sure.

The latest version of this PR can resize perfectly smoothly. And sometimes it jitters a little. And some times (like fyne_demo -> windows) it flashes red :(

File: internal/driver/glfw/window.go [hunk 2]
```diff
@@ -470,14 +466,18 @@ func (w *window) frameSized(viewport *glfw.Window, width, height int) {
...
+		w.canvas.setTextureScale(texScale)
+		w.canvas.painter.SetOutputSize(width, height)
+	})
+
+	refreshWindow(w)
 }
 
 func (w *window) refresh(viewport *glfw.Window) {
-	refreshWindow(w)
+	//	refreshWindow(w) // this seems to cause flashes of partial drawn canvas but I have no idea why... do we need it?
```
Comment (toaster) [inline internal/driver/glfw/window.go]: If this is not needed, it should be removed before merge.

**Comment by stuartmscott** [comment]
Looks good, before I approve please resolve conflicts

**Pull request merged**